python/foxglove-sdk/python/foxglove/notebook/FoxgloveViewer.py

```diff
@@ -0,0 +1,155 @@
+    height = traitlets.Unicode("500px").tag(sync=True)
+    src = traitlets.Unicode("").tag(sync=True)
+    layout_data = traitlets.Dict({}).tag(sync=True)
+    # A mapping of Context to NotebookBuffer, shared across all instances
+    _notebook_buffers_by_context: dict[Context, NotebookBuffer] = {}
+
+    def __init__(
+        self,
+        context: Optional[Context] = None,
+        width: Optional[str] = None,
+        height: Optional[str] = None,
+        src: Optional[str] = None,
+        layout_data: Optional[dict] = None,
+        **kwargs: Any,
+    ):
+        """
+        Initialize the FoxgloveViewer widget with the specified data source and configuration.
+
+        Args:
+            context: The Context used to log the messages. If no Context is provided, the global
+                context will be used. The visualization data will be retrieved from the
+                NotebookBuffer associated with the provided context. This buffer should have been
+                populated with logged messages before creating the viewer.
+            width: Optional width for the widget. Can be specified as CSS values like
+                "800px", "100%", "50vw", etc. If not provided, defaults to "100%".
+            height: Optional height for the widget. Can be specified as CSS values like
+                "600px", "80vh", "400px", etc. If not provided, defaults to "500px".
+            src: Optional URL of the Foxglove app instance to use. If not provided or empty,
+                uses the default Foxglove embed server (https://embed.foxglove.dev/).
+            layout_data: Optional layout data to be used by the Foxglove viewer. Should be a
+                dictionary that was exported from the Foxglove app. If not provided, uses the
+                default layout.
+            **kwargs: Additional keyword arguments passed to the parent AnyWidget class.
+
+        Note:
+            The widget will automatically load the data from the provided datasource
+            and display it in the embedded Foxglove viewer. The data is loaded once
+            during initialization - to update the data, use the :meth:`set_datasource` method.
+
+        Example:
+            >>> import foxglove
+            >>> foxglove.create_notebook_buffer()
+            >>> # ... log some data ...
+            >>> viewer = FoxgloveViewer(
+            ...     width="800px",
+            ...     height="600px"
+            ... )
+        """
+        super().__init__(**kwargs)
+
+        if width is not None:
+            self.width = width
+        if height is not None:
+            self.height = height
+        if src is not None:
+            self.src = src
+        if layout_data is not None:
+            self.layout_data = layout_data
+
+        # Use default context if no context is provided
+        ctx = context or Context.default()
+        self.set_data_from_context(ctx)
+
+    def set_data_from_context(self, context: Context) -> None:
```

Comment:
Is the user meant to call this themselves? It kinda looks like an internal method -- in which case I think it could be prefixed with `_` and we don't need the detailed docstring unless you find it helpful for internal use.

Reply:
Yeah, users should be able to call this method if they want to update the data in the viewer.